app/Domains/RevenueStreams/Actions/UpdateRevenueStream.php

```diff
@@ -30,15 +32,26 @@ public function handle(RevenueStream $revenue_stream, array $params): RevenueStr
         return $revenue_stream;
     }
 
-    public function rules(): array
+    public function rules(ActionRequest $request): array
     {
-        return [
+        $rules = [
             'id' => ['required', 'exists:revenue_streams,id'],
             'name' => ['sometimes', 'string', 'max:255'],
             'description' => ['sometimes', 'nullable', 'string'],
             'type_id' => ['sometimes', 'exists:revenue_stream_types,id'],
-            'values' => ['sometimes', 'json'],
+            'values' => ['sometimes', 'array'],
         ];
+
+        if ($request->has('type_id')) {
```

Comment:
Move this logic to `withValidator`, check [documentation](https://www.laravelactions.com/2.x/add-validation-to-controllers.html#custom-validation-logic) of `withValidator`.